src/rgw/rgw_acl_s3.cc

```diff
@@ -366,6 +366,11 @@ int RGWAccessControlList_S3::create_canned(ACLOwner& owner, ACLOwner& bucket_own
   rgw_user bid = bucket_owner.get_id();
   string bname = bucket_owner.get_display_name();
 
+  if (owner.get_id() == rgw_user(RGW_USER_ANON_ID)) {
+    owner.set_id(bid);
+    owner.set_name(bname);
```

Comment:
for PUT Object requests, `create_canned()` is getting a reference to `s->owner` in `ACLOwner& owner`, and we shouldn't mutate that here. this should either operate on a local copy, or we should change `create_canned()` to take owner by value instead

Reply:
the `owner_grant.set_canon` do not change the input. So I think we can just make a if branch between `owner_grant.set_canon(bid, bname, RGW_PERM_FULL_CONTROL);` and `owner_grant.set_canon(owner.get_id(), owner.get_display_name(), RGW_PERM_FULL_CONTROL);`